### ISSUE TITLE
Large indexes

### ISSUE THREAD

**Issue by jimmarshall87** [jerryjliu/llama_index#3516]
We are using the `GPTVectorStoreIndex` and begin to hit some practical limits of using it since it is all stored in memory. E.g. loading a 7000 article index in our case needs 3GB RAM.

If we use a Pinecone (or similar) backend with LlamaIndex on top, will this avoid the issue by never loading the full index into memory and just dynamically querying what it needs from the vector database, or does LlamaIndex still attempt to load the whole index into memory?

If the latter any recommendations on how to keep the memory footprint down?

**Comment by Disiok** [comment]
Yes, if you use an external vector store service, only the top-k documents will be loaded into memory.

**Comment by jimmarshall87** [comment]
Many thanks!
